assets/signing/sign-message.bouncycastle.cs

```diff
@@ -0,0 +1,73 @@
+// #########################################################
+// #             WARNING   WARNING   WARNING               #
+// #########################################################
+// #                                                       #
+// # This file is intended for demonstration purposes      #
+// # only.                                                 #
+// #                                                       #
+// # It is the SOLE responsibility of YOU, the programmer  #
+// # to prevent against unauthorized access to any signing #
+// # functions.                                            #
+// #                                                       #
+// # Organizations that do not protect against un-         #
+// # authorized signing will be black-listed to prevent    #
+// # software piracy.                                      #
+// #                                                       #
+// # -QZ Industries, LLC                                   #
+// #                                                       #
+// #########################################################
+
+using System;
+using System.Text;
+using Org.BouncyCastle.Crypto;
+using Org.BouncyCastle.Security;
+using Org.BouncyCastle.Utilities.IO.Pem;
+
+namespace QZTray
+{
+    public class QZTraySigning
+    {
+
+        // Public method for signing the input string with the private key
+        // ===============================================================
+
+        // QZTray.key is a PEM-encoded private key of "QZTray.crt".
+        //
+        //     I.e. It's the key of the CSR (Certificate Signing Request), which is the
+        //          original certificate before being signed by the CA key.
+        private string _key = "QZTray.key";
```

Comment:
We do not offer a namespace as it implies our code is a drop-in, which is is not.  Our examples all use `private-key.pem`, this consistency matches our wiki and youtube and cannot change. Minor but we also do not use `_underscore` variable notation as part of our examples.  Next, although the CSR statement is correct (and we support this through our portal) we no longer make mention of CSRs in our signing.  Finally, "PEM encoded" is a bit vague, we would recommend the comment is updated to specify PKCS#8, assuming that format works.  This is intentional on our end as PKCS#1 is a similar format, but breaks many of our examples and enforcing consistency ensures we can continue to support these snippets for years to come.

Reply:
> Finally, "PEM encoded" is a bit vague, we would recommend the comment is updated to specify PKCS#8, assuming that format works.

I had to replace `PemReader` class from `Org.BouncyCastle.Utilities.IO.Pem` with `PemReader` class from `Org.BouncyCastle.OpenSsl` as the latter complies with `PKCS #8` standard.